tests/Integration/SDK/SpanBuilderTest.php

```diff
@@ -230,6 +236,62 @@ public function test_set_attribute_after_end(): void
         $span->end();
     }
 
+    public function test_set_attribute_null_string_value(): void
+    {
+        /** @var Span $span */
+        $span = $this
+            ->tracer
+            ->spanBuilder(self::SPAN_NAME)
+            ->setAttribute('nil', null)
+            ->setAttribute('empty-string', '')
+            ->startSpan();
+
+        $attributes = $span->toSpanData()->getAttributes();
+        $this->assertSame(1, $attributes->count());
+        $this->assertSame('', $attributes->get('empty-string'));
+        $this->assertNull($attributes->get('nil'));
```

Comment:
this is a little confusing - did it get the nil attribute, or was null returned as a default for an undefined attribute?

Reply:
> this is a little confusing - did it get the nil attribute, or was null returned as a default for an undefined attribute?

It did not accept the nil attribute